rewrite-java/src/main/java/org/openrewrite/java/AddImport.java

```diff
@@ -151,23 +147,37 @@ public AddImport(String type, @Nullable String statik, boolean onlyIfReferenced)
                     .map(JavaSourceSet::getClasspath)
                     .orElse(Collections.emptyList());
 
-            cu = cu.getPadding().withImports(layoutStyle.addImport(cu.getPadding().getImports(), importToAdd,
-                    cu.getPackageDeclaration(), classpath));
+            List<JRightPadded<J.Import>> newImports = layoutStyle.addImport(cu.getPadding().getImports(), importToAdd, cu.getPackageDeclaration(), classpath);
+
+            // ImportLayoutStile::addImport adds always `\n` as newlines. Checking if we need to fix them
+            newImports = checkCRLF(cu, newImports);
+
+            cu = cu.getPadding().withImports(newImports);
 
             JavaSourceFile c = cu;
-            cu = cu.withClasses(ListUtils.map(cu.getClasses(), (i, clazz) -> {
-                if (i == 0) {
-                    J.ClassDeclaration cl = autoFormat(clazz, clazz.getName(), p, new Cursor(null, c));
-                    clazz = clazz.withPrefix(clazz.getPrefix().withWhitespace(cl.getPrefix().getWhitespace()));
-                }
-                return clazz;
+            cu = cu.withClasses(ListUtils.mapFirst(cu.getClasses(), clazz -> {
+                J.ClassDeclaration cl = autoFormat(clazz, clazz.getName(), p, new Cursor(null, c));
+                return clazz.withPrefix(clazz.getPrefix().withWhitespace(cl.getPrefix().getWhitespace()));
             }));
 
             j = cu;
         }
         return j;
     }
 
+    private List<JRightPadded<J.Import>> checkCRLF(JavaSourceFile cu, List<JRightPadded<J.Import>> newImports) {
+        GeneralFormatStyle generalFormatStyle = Optional.ofNullable(((SourceFile) cu).getStyle(GeneralFormatStyle.class))
+                .orElse(autodetectGeneralFormatStyle(cu));
+        if (generalFormatStyle.isUseCRLFNewLines()) {
+            return ListUtils.map(newImports, rp -> rp.map(
+                    i -> i.withPrefix(i.getPrefix().withWhitespace(i.getPrefix().getWhitespace()
+                            .replace("\n", "\r\n")
+                            .replaceAll("\r+", "\r")))
```

Comment:
You could do this in one shot by using a negative look behind.

```suggestion
                            .replaceAll("(?<!\r)\n", "\r\n")))
```

Reply:
thanks! :)